src/api.c

```diff
@@ -1321,6 +1321,60 @@ static void cgroup_free_cg_mount_table(void)
 	memset(&cg_cgroup_v2_empty_mount_paths, 0, sizeof(cg_cgroup_v2_empty_mount_paths));
 }
 
+/*
+ * Parses the mount options of the given mount point and checks for the
+ * option in the list of mount options.
+ * @mnt: Mount point name to search for mount points.
+ * @mnt_option: Mount option to be searched.
+ *
+ * Returns 0, in case of success and ECGOTHER on failure.
+ */
+static int check_mount_point_opt(char *mnt, char *mnt_option)
+{
```

Comment:
It feels unintuitive to return `0` for "yes, this opt is set" and an error code for all other cases.  If it's not set, the user is still getting an error; should the user abort or does the error simply mean the opt isn't set?

Would it make more sense to pass in an `is_set` parameter that this function could write to?
```
static int check_mount_point_opt(char *mnt, char *mnt_option, int * const is_set)
```

Reply:
A clearer way to state what I wrote above.  I think there should be three clear returns from this function:
* `mnt_option` WAS found
* `mnt_option` WASN'T found
* error